### PR TITLE
generate: added --no-strict-mode flag

### PR DESCRIPTION
**Pull request by kolmar**
Changes:

* Added flag `--no-strict-mode` to `execute-generators`
* Bumped backend version to 0.12

README changes:

* Reformatted README.md for execute-generators with `COLUMNS=100 execute-generators -h`

### CODE REVIEW

File: execute-generators/README.md [hunk 1]
```diff
@@ -27,45 +24,45 @@ required arguments:
...
+  --exclude-module EXCLUDE_MODULE         list of modules to exclude from generation
+
+  --parallel-generation-threads THREADS   Number of threads to use for parallel generation. Value
+                                          of 0 means that parallel generation is disabled.
+                                          Default: 0
 
-error exit codes:
-  254: nothing to generate
-  255: general MPS error
+  --disable-strict-mode                   Disable strict generation mode. Strict mode places
```
Comment (sergej-koscejev) [inline execute-generators/README.md:56]: Let's call it `--no-strict-mode` for consistency with other command-line utilities.

File: integration-tests/build.gradle.kts [hunk 2]
```diff
@@ -28,6 +28,9 @@ val GENERATION_TESTS = listOf(
     GenerationTest("generateBuildSolutionParallel", "generate-build-solution",
         listOf("--model", "my.build.script", "--parallel-generation-threads=4"),
         expectation = GenerationTestExpectation.SuccessWithFiles("solutions/my.build/build.xml")),
+    GenerationTest("generateBuildSolutionNonStrict", "generate-build-solution",
+        listOf("--model", "my.build.script", "--disable-strict-mode"),
```
Comment (sergej-koscejev) [inline integration-tests/build.gradle.kts:32]: Let's skip this test.

We should later change the tests for generation settings to just test the settings without running MPS. The more integration tests we have, the slower they run.

File: execute-generators/src/main/kotlin/de/itemis/mps/gradle/generate/GenerateArgs.kt [hunk 3]
```diff
@@ -12,8 +12,15 @@ class GenerateArgs(parser: ArgParser) : Args(parser) {
...
+        help = "Number of threads to use for parallel generation. Value of 0 means that parallel generation is disabled. Default: 0",
+        argName = "THREADS"
     ) { toInt() }
         .default(0)
         .addValidator { if (value < 0) throw InvalidArgumentException("parallel-generation-threads must be >= 0") }
+    val disableStrictMode by parser.flagging(
+        "--disable-strict-mode",
+        help = "Disable strict generation mode. Strict mode places additional limitations on generators, but is required for parallel generation"
+    ).addValidator {
+        if (parallelGenerationThreads != 0 && value) throw InvalidArgumentException("strict mode is required for parallel generation")
```
Comment (sergej-koscejev) [inline execute-generators/src/main/kotlin/de/itemis/mps/gradle/generate/GenerateArgs.kt:24]: Is this validation dependent on the order of flags on the command line?

Also, in my mind it belongs on the parallel-generation-threads option.

File: execute-generators/README.md [hunk 4]
```diff
@@ -11,14 +11,11 @@ is by using Gradle's `JavaExec` task. See below for an example.
...
-                          [--parallel-generation-threads THREADS]
-                          [--log-level LOG_LEVEL] [--model MODEL]...
-                          [--module MODULE]...
-                          [--exclude-model EXCLUDE_MODEL]...
-                          [--exclude-module EXCLUDE_MODULE]...
+                          [--plugin-location PLUGIN_LOCATION] [--build-number BUILD_NUMBER]
+                          --project PROJECT [--test-mode] [--environment ENVIRONMENT]
+                          [--log-level LOG_LEVEL] [--model MODEL]... [--module MODULE]...
+                          [--exclude-model EXCLUDE_MODEL]... [--exclude-module EXCLUDE_MODULE]...
+                          [--parallel-generation-threads THREADS] [--disable-strict-mode]
```
Comment (sergej-koscejev) [inline execute-generators/README.md:18]: 100 columns were a good idea! 👍

**Review by sergej-koscejev — CHANGES_REQUESTED** [review]

File: execute-generators/src/main/kotlin/de/itemis/mps/gradle/generate/GenerateArgs.kt (hunk 3 above)
Comment (kolmar) [reply execute-generators/src/main/kotlin/de/itemis/mps/gradle/generate/GenerateArgs.kt]: Independent. All validations run after parsing all args.
Moved.

File: integration-tests/build.gradle.kts (hunk 2 above)
Comment (kolmar) [reply integration-tests/build.gradle.kts]: Removed

File: execute-generators/README.md (hunk 1 above)
Comment (kolmar) [reply execute-generators/README.md]: Renamed

**Review by sergej-koscejev — APPROVED** [review]

**Pull request merged**